### DISCUSSION TITLE
Policy to restrict access to one column?

### DISCUSSION THREAD

**Discussion by Etheonor** [supabase/supabase#1128] [Questions]
I'm sure it's a postres related question but I want to know if we can add a specific policy in the supabase dashboard to lock access (read/write to a specific column?)

For example, in my public/users table I want the users to be able to read/edit their nickname, social accounts etc... but if I also want to add a "stripe ID" number, is there a way to restrict access to a specific column?

**Comment by silentworks** [comment]
We don't normally tend to lock access to columns in relational databases, it's normally better to structure your tables differently. I would suggest moving the stripe ID column to a different table and create a relationship to the `public.users` table to it. Also if this is a server side application then you can just make sure that the object you pass to the supabase.update function only contain the columns that you want updated.

Otherwise I found this solution on Stack Overflow that you can use to throw an error if someone tries to update a certain column https://stackoverflow.com/a/44301987

**Comment by inian** [comment]
Also check out the previous discussion answered by @steve-chavez [here](https://github.com/supabase/supabase/discussions/656#discussioncomment-335362)
